### PR TITLE
Ako/ add gb feature flag

### PR DESCRIPTION
**Pull request by ali-hosseini-deriv**
This is to use Growthbook featureflag to redirect back the clients to deriv-app if the flag is disabled.

### CODE REVIEW

File: src/hooks/custom-hooks/useIsGrowthbookLoaded.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import { Analytics } from '@deriv-com/analytics';
+
+const useIsGrowthbookIsLoaded = () => {
+    const [isGBLoaded, setIsGBLoaded] = useState(false);
+
+    useEffect(() => {
+        let checksCounter = 0;
+        const analyticsInterval: NodeJS.Timeout = setInterval(() => {
+            // Check if the analytics instance is available for 10 seconds before setting the feature flag value
+            if (checksCounter > 20) {
```
Comment (matin-deriv) [inline src/hooks/custom-hooks/useIsGrowthbookLoaded.tsx:11]: Why do we specifically mention about the interval time? is it 10 or 20?

File: src/App.tsx [hunk 2]
```diff
@@ -23,6 +28,13 @@ const App = () => {
     initDerivAnalytics();
     redirectToOauth();
 
+    useEffect(() => {
+        if (isGBLoaded) {
+            if (!isP2PV2Enabled) {
+                window.location.href = 'https://app.deriv.com/cashier/p2p';
```
Comment (matin-deriv) [inline src/App.tsx:34]: I don't think we should hardcode this in the code? how about Utils?

File: src/hooks/custom-hooks/useIsGrowthbookLoaded.tsx (hunk 1 above)
Comment (ali-hosseini-deriv) [reply src/hooks/custom-hooks/useIsGrowthbookLoaded.tsx:11]: It's 10 seconds.
500ms * 20 (Number of Checks) = 10seconds

Comment (ali-hosseini-deriv) [reply src/hooks/custom-hooks/useIsGrowthbookLoaded.tsx:11]: and we just need to check this for 10 seconds as its not clear when the Growthbook is available on clients machine, and its based on their network speed.

File: src/App.tsx (hunk 2 above)
Comment (ali-hosseini-deriv) [reply src/App.tsx:34]: You are right, Let me see where can i put it.

File: src/App.tsx [hunk 3]
```diff
@@ -7,13 +7,18 @@ import { useDerivAnalytics, useRedirectToOauth, useTrackjs } from '@/hooks';
...
+import { URLConstants } from '@deriv-com/utils';
+import useGrowthbookGetFeatureValue from './hooks/custom-hooks/useGrowthbookGetFeatureValue';
 
 const { VITE_CROWDIN_BRANCH_NAME, VITE_PROJECT_NAME, VITE_TRANSLATIONS_CDN_URL } = process.env;
 const i18nInstance = initializeI18n({
     cdnUrl: `${VITE_TRANSLATIONS_CDN_URL}/${VITE_PROJECT_NAME}/${VITE_CROWDIN_BRANCH_NAME}`,
 });
 
 const App = () => {
+    const [ShoudlRedirectToDerivApp, isGBLoaded] = useGrowthbookGetFeatureValue({
```
Comment (farrah-deriv) [inline src/App.tsx:19]: i don't think `ShoudlRedirectToDerivApp` is being returned

Comment (ali-hosseini-deriv) [reply src/App.tsx:19]: what do you mean by that @farrah-deriv ?
didn't get that.

Comment (farrah-deriv) [reply src/App.tsx:19]: i can't see `ShoudlRedirectToDerivApp` being returned from `useGrowthbookGetFeatureValue` hook

Comment (ali-hosseini-deriv) [reply src/App.tsx:19]: that's the hook implementation, you can name it whatever is suitable for you.

Comment (ali-hosseini-deriv) [inline src/App.tsx:19]:
```suggestion
    const [ShouldRedirectToDerivApp, isGBLoaded] = useGrowthbookGetFeatureValue({
```
Typo :-D 
fixing it.

File: src/App.tsx [hunk 4]
```diff
@@ -23,6 +28,13 @@ const App = () => {
     initDerivAnalytics();
     redirectToOauth();
 
+    useEffect(() => {
+        if (isGBLoaded && ShoudlRedirectToDerivApp) {
```
Comment (ali-hosseini-deriv) [inline src/App.tsx:32]:
```suggestion
        if (isGBLoaded && ShouldRedirectToDerivApp) {
```

File: src/App.tsx [hunk 5]
```diff
@@ -23,6 +28,13 @@ const App = () => {
     initDerivAnalytics();
     redirectToOauth();
 
+    useEffect(() => {
+        if (isGBLoaded && ShouldRedirectToDerivApp) {
+            const NODE_ENV = process.env.VITE_NODE_ENV;
+            const APP_URL = NODE_ENV === 'production' ? URLConstants.derivAppProduction : URLConstants.derivAppStaging;
+            window.location.href = `${APP_URL}/cashier/p2p`;
+        }
+    }, [isGBLoaded, ShoudlRedirectToDerivApp]);
```
Comment (ali-hosseini-deriv) [inline src/App.tsx:37]:
```suggestion
    }, [isGBLoaded, ShouldRedirectToDerivApp]);
```

**Pull request merged**